src/goto-programs/remove_function_pointers.cpp

```diff
@@ -521,6 +538,110 @@ void remove_function_pointerst::remove_function_pointer(
 
 /*******************************************************************\
 
+Function: remove_function_pointerst::does_program_contain_const_removal_cast
```

Comment:
This doesn't seem to be specific to function pointers. I would thus opt to have it as a separate analysis, just like dirtyt or is_threaded.